src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENT_DATA = {};
```

Comment:
Why do you need to have this? Your array should never be empty. How will you change a capacity if array will be empty?

Reply:
in the ArrayList class, when an instance is created through an empty constructor, an empty array is created and when the first element is added, the array increases to the default value due to the call to the 'grow' method, so I did the same.